### DISCUSSION TITLE
What threshold should I set on a new project?

### DISCUSSION THREAD

**Discussion by CraigComeOnThisNameCantBeTaken** [stryker-mutator/stryker-net#2416] [Q&A] [answered]
I understand that this is a personal decision but I am considering setting a break threshold of 100% on a new project because I want to keep standards high, and can accept an overhead for disabling some files and lines from consideration. But is that a good idea? Is anyone else doing that please?

**Comment by rouke-broersma** [comment]
I don't see any reason not to try, but reconsider if you notice it's not working for you :)

**Comment by dupdob** [comment, marked as answer]
Hi Craig.
I do. My target is at 1000 % for [NFluent](http://www.n-fluent.net/). I have been focused on Sryker for the past months, but last time I checked, I was around 99.xx% (I will circle back later why not 100%).
I did a post on my journey toward 100% (see [here](https://medium.com/@Cyrdup/mutation-testing-960759c44c0?source=friends_link&sk=7ba6e359c192ee3241f69b848c2addf1) ).

Here are my key suggestions for this.

1. Focus on getting 100% coverage first. Line coverage is usually sufficient. Why: because you can't kill a mutation you do not cover
2. Try to revise existing tests (such as adjusting the parameters/prepare phase), but accept that you will have to write supplemental tests to kill mutants
3. Immortal mutants: some mutants can not be killed. 
    1. If it is a 'semantically equivalent mutation', i.e. the mutant does not actually change what the code does, you must ignore it via Stryker comments.
    2. If the mutation does not alter the behavior of your code (paragraph 4 in my post), consider improving your design, and if you cannot improve it any further, ignore the mutation with Stryker comments
4. Unvoverable code: some code is sometime non coverable because it depends on some external components/services which is difficult to mock. Then you need to mark those part as non coverable, using `[ExcludeFromCodeCoverage]`. I would advise to put all related code to a dedicated project/assembly to ensure your core logic is still 100% covered


Hoping you will find any of this helpful. Wishing you the best on your project.
